src/components/table/AppPaginatedTable.vue

```diff
@@ -58,4 +55,11 @@ defineProps<{
   selectable?: boolean;
   rowClass?: (item: I) => string;
 }>();
+
+// Slots are passed to AppTable, typing is currently lost
```

Comment:
Why can you not use types here?

Reply:
Basically because of the way I am passing the slots down to `AppTable`.

`AppTable` has clearly defined slots, with a type that it something like
```ts
Partial<Record<`header-${string}`|`value-${string}`|"empty"|"loading", (_: {}) => any>>
```
This comes from the Vue compiler which sees that in `AppTable.vue` we create slots for those template strings.

However `AppPaginatedTable` just passes any slots it is given to `AppTable`, it doesn't specify `header-${string}` etc. So the list of slot names is any string, i.e.
```ts
const slotNames = Object.keys(slots);
// Type of slotNames is string[]
// But technically should be (`header-${string}`|`value-${string}`|"empty"|"loading")[]
```

The longer term fix is to figure out how to make `AppPaginatedTable` properly typed, but that's a bit beyond this PR :)

Reply:
okay thank you ;)